CMSgov/design-system
# [WNMGS-437] Move doc files to @cmsgov/design-system-docs

### Summary
- All doc files (_docs.scss, .example.html, .example.jsx) are moved into @cmsgov/design-system-docs so that all documentation related files are located in one place (with the exception of React props)
- Doc site generation for react props and react examples have been changed to be more explicit, `@react-component` previously would add both react examples and react props, even though there is another tag called `@react-example`. This PR replaces `@react-component` with `@react-props`, which is supplied with the name of the React file where props come from.
- `@hide-example` tag was removed, instead simplify don't specify a `@react-example` tag.

How to test:
- Follow same steps [here](https://github.com/CMSgov/design-system/pull/659)
- For child DS, use this 
[child-design-system-example 2.zip](https://github.com/CMSgov/design-system/files/4548680/child-design-system-example.2.zip)
- Additionally, test the "View source file" links for both core and child DS react files. Note: the links will 404 because the changes arent merged with master yet, but just confirm that they will work
